generator/internal/client_generator.cc

```diff
@@ -126,14 +126,23 @@ class $client_class_name$ {
   ///@}
 )""");
 
+  std::string const deprecation_macro_str =
+      R"""(  GOOGLE_CLOUD_CPP_DEPRECATED("This RPC is deprecated.")
+)""";
+
   for (google::protobuf::MethodDescriptor const& method : methods()) {
+    bool is_method_deprecated =
+        method.options().has_deprecated() && method.options().deprecated();
+    std::string const deprecation_macro =
+        is_method_deprecated ? deprecation_macro_str : "";
     if (IsBidirStreaming(method)) {
       HeaderPrintMethod(
           method, __FILE__, __LINE__,
           absl::StrCat(
               "\n",
               FormatMethodComments(method, "", IsDiscoveryDocumentProto()),
               // clang-format off
+              deprecation_macro,
```

Comment:
It might be clever-er to put the `deprecation_macro` at the end of `FormatMethodComments()`.

Con:
- The deprecation macro is not a comment....

Pro:
- `FormatMethodComments()` already knows whether a `method` is deprecated or not.

Reply:
Yeah, I think about adding `deprecation_macro` to `FormatMethodComments()` at the beginning. But there will be some rare cases, like if there are some methods without comments, then we will miss the deprecation tag.

Also, deprecation tag is part of the method rather than comments.

I am not very certain whether I should add it to `FormatMethodComments()`, please let me know if you think that is the right way, I will modify it.

Reply:
> like if there are some methods without comments

All of our client methods have comments.

> Also, deprecation tag is part of the method rather than comments.

Yeah. The naming is why I was iffy.

My biggest concern is about missing a case. Our golden files do not cover every type of deprecated RPC. (We only test the unary case. We do not test cases like streams, or async methods, etc.).

I can see that you covered all the cases in the code. So I am cool with the PR as is. I don't think we would notice if one of those cases got removed though. :shrug: